src/providers/microsoft/azure/mod.rs

```diff
@@ -402,18 +405,16 @@ impl MetadataProvider for Azure {
         let goalstate = self.fetch_goalstate()?;
         let certs_endpoint = match goalstate.certs_endpoint() {
             Some(ep) => ep,
-            None => {
-                warn!("SSH pubkeys requested, but not provisioned for this instance");
-                return Ok(vec![]);
-            }
+            None => return Ok(vec![]),
         };
 
         if certs_endpoint.is_empty() {
             bail!("unexpected empty certificates endpoint");
         }
 
-        let key = self.get_ssh_pubkey(certs_endpoint)?;
-        Ok(vec![key])
+        let key: Vec<PublicKey> = self.get_ssh_pubkey(certs_endpoint)?.unwrap_or(vec![]);
+
+        Ok(key)
```

Comment:
Then this code here would remain untouched.

IOW, the lower-level `get_ssh_pubkey()` function can just focus on doing the obvious thing and this function here is where we have to deal with what our callers expect from us.

Reply:
I was having a difficult time figuring out how to return a vector on `None` without inspecting the Option like 
https://github.com/coreos/afterburn/blob/c3aa867278ab4c89a6a45a3a5ee239d31ee4509e/src/providers/microsoft/azure/mod.rs#L406-L408

Reply:
Oh right sorry, I missed that detail here. There's a neat Rust trick which is that `Option` actually supports iteration, in which case it'll iterate none or once (if the item exists). So something like `Ok(maybe_key.into_iter().collect())` should work.

Reply:
Perfect, thanks! I made that change.